Assets/Scripts/Source/Game/Combat/CombatManager.cs

```diff
@@ -59,9 +83,122 @@ void generateEnemyActions()
     }
 
     IEnumerator executeActions()
-    {   
+    {
         while(!turnManager.ShouldWaitForPlayerAction(characterManager.GetActivePlayer()))
         {
+            float timer = 0.0f;
+            float missDelay = 0.0f;
+            bool missed = false;
+            CharacterAction action = turnManager.Peek();
+            CharacterAction enemyAction = action.Targets[0].ActiveAction; //TODO: Maybe add target enemy incase of multiple enemies
+            
+            if (action.Actor is Player)
```

Comment:
Can we break these into helper functions? 

`handlePlayerAction` --> `handlePlayerAttack`